contracts/JBController.sol

```diff
@@ -370,25 +371,93 @@ contract JBController is IJBController, JBTerminalUtility, JBOperatable, Reentra
+      @dev _metadata.reservedRate A number from 0-10000 (0-100%) indicating the percentage of each contribution's newly minted tokens that will be reserved for the token splits.
+      @dev _metadata.redemptionRate The rate from 0-10000 (0-100%) that tunes the bonding curve according to which a project's tokens can be redeemed for overflow.
+        The bonding curve formula is https://www.desmos.com/calculator/sp9ru6zbpk
+        where x is _count, o is _currentOverflow, s is _totalSupply, and r is _redemptionRate.
+      @dev _metadata.ballotRedemptionRate The redemption rate to apply when there is an active ballot.
+      @dev _metadata.pausePay Whether or not the pay functionality should be paused during this cycle.
+      @dev _metadata.pauseWithdrawals Whether or not the withdraw functionality should be paused during this cycle.
+      @dev _metadata.pauseRedeem Whether or not the redeem functionality should be paused during this cycle.
+      @dev _metadata.pauseMint Whether or not the mint functionality should be paused during this cycle.
+      @dev _metadata.pauseBurn Whether or not the burn functionality should be paused during this cycle.
+      @dev _metadata.allowTerminalMigration Whether or not the terminal migration functionality should be paused during this cycle.
+      @dev _metadata.allowControllerMigration Whether or not the controller migration functionality should be paused during this cycle.
+      @dev _metadata.holdFees Whether or not fees should be held to be processed at a later time during this cycle.
+      @dev _metadata.useDataSourceForPay Whether or not the data source should be used when processing a payment.
+      @dev _metadata.useDataSourceForRedeem Whether or not the data source should be used when processing a redemption.
+      @dev _metadata.dataSource A contract that exposes data that can be used within pay and redeem transactions. Must adhere to IJBFundingCycleDataSource.
+    @param _mustStartAtOrAfter The time before which the configured funding cycle can't start.
+    @param _groupedSplits An array of splits to set for any number of group.
+    @param _fundAccessConstraints An array containing amounts, in wei (18 decimals), that a project can use from its own overflow on-demand for each payment terminal.
+    @param _terminals Payment terminals to add for the project.
+
+    @return configuration The configuration of the funding cycle that was successfully created.
+   */
+
+  function launchFundingCycleFor(
+    uint256 projectId,
+    JBFundingCycleData calldata _data,
+    JBFundingCycleMetadata calldata _metadata,
+    uint256 _mustStartAtOrAfter,
+    JBGroupedSplits[] memory _groupedSplits,
+    JBFundAccessConstraints[] memory _fundAccessConstraints,
+    IJBTerminal[] memory _terminals
+  ) external returns (uint256 configuration) {
+    // Check the project exists and is owned by msg.sender
+    if (msg.sender != projects.ownerOf(projectId)) revert CALLER_NOT_PROJECT_OWNER();
```

Comment:
this can use a `requiresPermission` modifier. see `reconfigureFundingCyclesFor` tx